src/interp.rs

```diff
@@ -512,7 +322,7 @@ mod specialized {
             }
         }
 
-        pub(super) fn push(&mut self, frame: &F) -> [F; FACTOR2] {
+        pub(crate) fn push(&mut self, frame: &F) -> [F; FACTOR2] {
```

Comment:
As this is the only function really left here, maybe `interpolate()`, `convert()`, `resample()`, or similar would work better? In the end, this resamples on frame to 2 or 4 output frames.